### ISSUE TITLE
"Publish on .." seems broken. 

### ISSUE THREAD

**Issue by bobdenotter** [bolt/bolt#5733]
If you set a publication date on an entry, it gets wiped after publishing. 

Details
-------

 - Relevant Bolt Version: [ master ]

Reproduction
------------

Set the publication date to something in the near future:

![screen shot 2016-09-05 at 20 40 44](https://cloud.githubusercontent.com/assets/1833361/18255441/ed283e1a-73a9-11e6-8789-7c1428f8a7bd.png)

Get an exception, when that future date happens: 

![screen shot 2016-09-05 at 20 41 21](https://cloud.githubusercontent.com/assets/1833361/18255445/f355bd6c-73a9-11e6-9467-058fe85ca55f.png)

Return to the entry, to see it's gone: 

![screen shot 2016-09-05 at 20 41 40](https://cloud.githubusercontent.com/assets/1833361/18255446/f7624d4e-73a9-11e6-8331-e636272d1a92.png)

**Comment by bobdenotter** [comment]
Full(er) stack trace: 

```
in FileCache.php line 271
at RecursiveDirectoryIterator->__construct('/Users/bobdenotter/Sites/bolt-origin/app/cache/development/data', '4096') in FileCache.php line 271
at FileCache->getIterator() in FileCache.php line 175
at FileCache->doFlush() in Cache.php line 68
at Cache->doFlush() in CacheProvider.php line 156
at CacheProvider->flushAll() in Connection.php line 76
at Connection->executeUpdate('UPDATE bolt_taxonomy SET id = :id, content_id = :content_id, contenttype = :contenttype, taxonomytype = :taxonomytype, slug = :slug, name = :name, sortorder = :sortorder WHERE id = :id', array('id' => '199', 'content_id' => '36', 'contenttype' => 'entries', 'taxonomytype' => 'categories', 'slug' => 'books', 'name' => 'books', 'sortorder' => '0'), array()) in QueryBuilder.php line 208
at QueryBuilder->execute() in QuerySet.php line 46
at QuerySet->execute() in Repository.php line 414
at Repository->update(object(Taxonomy)) in Repository.php line 357
at Repository->save(object(Taxonomy)) in TaxonomyType.php line 148
at TaxonomyType->Bolt\Storage\Field\Type\{closure}(object(QueryBuilder), '1', '36') in QuerySet.php line 51
at QuerySet->execute() in Repository.php line 414
at Repository->update(object(Content)) in Repository.php line 357
at Repository->save(object(Content)) in TimedRecord.php line 158
at TimedRecord->save(object(ContentRepository), object(Content), 'publish', null) in TimedRecord.php line 143
at TimedRecord->timedHandleRecords('entries', 'publish') in TimedRecord.php line 92
at TimedRecord->publishTimedRecords() in StorageEventListener.php line 118
at StorageEventListener->onKernelRequest(object(GetResponseEvent), 'kernel.request', object(TraceableEventDispatcher))
at call_user_func(array(object(StorageEventListener), 'onKernelRequest'), object(GetResponseEvent), 'kernel.request', object(TraceableEventDispatcher)) in WrappedListener.php line 61
at WrappedListener->__invoke(object(GetResponseEvent), 'kernel.request', object(EventDispatcher))
at call_user_func(object(WrappedListener), object(GetResponseEvent), 'kernel.request', object(EventDispatcher)) in EventDispatcher.php line 184
at EventDispatcher->doDispatch(array(object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'kernel.request', object(GetResponseEvent)) in EventDispatcher.php line 46
at EventDispatcher->dispatch('kernel.request', object(GetResponseEvent)) in TraceableEventDispatcher.php line 140
at TraceableEventDispatcher->dispatch('kernel.request', object(GetResponseEvent)) in HttpKernel.php line 125
at HttpKernel->handleRaw(object(Request), '1') in HttpKernel.php line 64
at HttpKernel->handle(object(Request), '1', true) in Application.php line 586
at Application->handle(object(Request)) in Application.php line 563
at Application->run(object(Request)) in Application.php line 88
at Application->run() in index.php line 14
```

**Comment by GawainLynch** [comment]
That is our caching messing up, we must be triggering something early still :disappointed:

**Comment by GawainLynch** [comment]
Being triggered [here](https://github.com/bolt/bolt/blob/release/3.2/src/Storage/Database/Connection.php#L76)

**Comment by bobdenotter** [comment]
Fixed by #5879
